### ISSUE TITLE
Errors for Reason

### ISSUE THREAD

**Issue by EduardoRFS** [melange-re/melange#166]
As we removed `super errors` at #48, `bsc` always reports errors in plain OCaml. Should we do anything to report errors in Reason?

### Integrating `refmterr`

- another external dependency
- improves formatting
- if ocaml changes formatting it will break
- doesn't work for all errors

### Hacking printtyp

- keeps formatting
- future proof
- works for all errors
- requires hooks on [printtyp.ml](https://github.com/melange-re/melange-compiler-libs/blob/4.12.0%2BBS/typing/printtyp.ml)

**Comment by EduardoRFS** [comment]
My opinion, we should add the hooks and try to upstream them, as it would allow to have custom `ocamlc` and `ocamlopt` quite easily. That also is capable of showing error messages in Reason.

But while this is not upstreamed it will still be a maintenance burden here.

**Comment by Lupus** [comment]
My VS Code shows me types and type errors in Reason syntax when I'm editing .re/.rei file. I'm using OCaml Platform with ocaml-lsp-server, may be there is some code already there that could be leveraged?

**Comment by EduardoRFS** [comment]
@Lupus yeah that's with merlin, which has patches for it, but not for the compiler messages, like when running `dune build -w @check`

**Comment by jfrolich** [comment]
Yeah, I believe having OCaml errors is not great for making ReasonML accessible. But something that can be upstreamed to OCaml eventually sounds like the most sustainable solution.

**Comment by EduardoRFS** [comment]
The hooks were actually already in place, which means no change was required on the compiler. @Lupus probably merlin is just setting the `Oprint.print____` references.

I think we could actually make a compiler wrapper for this to Reason Native

**Comment by jchavarri** [comment]
It seems that https://github.com/melange-re/melange/pull/186 fixed the issue originally, but after integrating deeper with dune, and replace ninja with it, the problem is back. cc @davesnx @sanderspies
